public/javascripts/Choropleth.js

```diff
@@ -1,241 +1,327 @@
      */
-
-    function getColor(p) {
-        //since this is a float, we cannot directly compare. Using epsilon to avoid floating point errors
-        return Math.abs(p - 100) < Number.EPSILON ? '#03152f':
-                 p > 90 ? '#08306b' :
-                    p > 80 ? '#08519c' :
-                        p > 70 ? '#08719c' :
-                            p > 60 ? '#2171b5' :
-                                p > 50 ? '#4292c6' :
-                                    p > 40 ? '#6baed6' :
-                                        p > 30 ? '#82badb' :
-                                            p > 20 ? '#9ecae1' :
-                                                p > 10 ? '#b3d3e8' :
-                                                    '#c6dbef';
+    function initializeOverlayPolygon(map, lat, lng) {
+        var overlayPolygon = {
+            "type": "FeatureCollection",
+            "features": [{"type": "Feature", "geometry": {
+                "type": "Polygon", "coordinates": [
+                        [
+                            [lng + 2, lat - 2],
+                            [lng + 2, lat + 2],
+                            [lng - 2, lat + 2],
+                            [lng - 2, lat - 2],
+                            [lng + 2, lat - 2]
+                        ]
+                ]}}]};
+        var layer = L.geoJson(overlayPolygon);
+        layer.setStyle({color: "#ccc", fillColor: "#ccc"});
+        layer.addTo(map);
     }
 
-    /**
-     * render the neighborhood polygons, colored by completion percentage
-     */
+    // Renders the neighborhood polygons, colored by completion percentage.
     function initializeChoroplethNeighborhoodPolygons(map, rates) {
-        var neighborhoodPolygonStyle = { // default bright red, used to check if any regions are missing data
-                color: '#888',
-                weight: 1,
-                opacity: 0.25,
-                fillColor: "#f00",
-                fillOpacity: 1.0
-            },
+        var regionData;
+         // Default region color, used to check if any regions are missing data.
+        var neighborhoodPolygonStyle = params.neighborhoodPolygonStyle,
             layers = [],
             currentLayer;
 
-        // finds the matching neighborhood's completion percentage, and uses it to determine the fill color
+        // Finds the matching neighborhood's completion percentage, and uses it to determine the fill color.
         function style(feature) {
-            for (var i = 0; i < rates.length; i++) {
-                if (rates[i].region_id === feature.properties.region_id) {
-
-                    return {
-                        color: '#888',
-                        weight: 1,
-                        opacity: 0.25,
-                        fillColor: getColor(100.0 * rates[i].rate),
-                        fillOpacity: 0.35 + (0.4 * rates[i].rate)
+            if (params.singleRegionColor) {
```

Comment:
We should add some parameter that determines the way we find the polgyons' style. It could also replace the `singleRegionColor` parameter. The options might be "issueCount", "completionRate", and "singleColor", and that will determine what function to call to get the correct coloring.